CRM/Contribute/BAO/Contribution.php

```diff
@@ -141,14 +140,19 @@ public static function add(&$params, $ids = []) {
         $params['creditnote_id'] = self::createCreditNoteId();
       }
     }
-    if (empty($params['contribution_status_id'])) {
+    $contributionStatusID = $params['contribution_status_id'] ?? NULL;
+    if (CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', (int) $contributionStatusID) === 'Partially paid' && empty($params['is_post_payment_create'])) {
+      CRM_Core_Error::deprecatedFunctionWarning('Setting status to partially paid other than by using Payment.create is deprecated and unreliable');
```

Comment:
If a contribution with completed status is edited so that the total amount increases, then the status is supposed to change to partially paid. So in that use case I think this error message is inappropriate.

Reply:
@JoeMurray we have quite a lot of testing on editing values and this line is never hit or the test would fail.

Reply:
@JoeMurray this seems safe to me as the changes that happen here https://github.com/civicrm/civicrm-core/pull/16218/files#diff-0474c991325eed0cec87e8240916ef79R507 which cause the total amount of a contribution linked to an event do not trigger this